libkbfs/folder_branch_ops.go

```diff
@@ -548,9 +548,21 @@ func (fbo *folderBranchOps) setHeadLocked(
 		// revision is enough to trigger conflict resolution.
 		fbo.cr.Resolve(md.Revision(), MetadataRevisionUninitialized)
 	} else if md.MergedStatus() == Merged {
-		// If we are already merged through this write, the revision would be the
-		// latestMergedRevision on server.
-		fbo.setLatestMergedRevisionLocked(ctx, lState, md.Revision(), false)
+		journalEnabled := TLFJournalEnabled(fbo.config, fbo.id())
+		var key kbfscrypto.VerifyingKey
+		if journalEnabled {
+			key, err = fbo.config.KBPKI().GetCurrentVerifyingKey(ctx)
+			if err != nil {
+				return err
+			}
+		}
+		// If this is a merged revision, it should be the latest one
+		// we know about.  However, if journaling is enabled, and this
```

Comment:
How is it possible for revisions not from this device to be in the journal? It seems to me that if the journal is enabled, we could just rely on `onMDFlush` to take care of it.